service/notesservice.php

```diff
@@ -209,12 +217,23 @@ private function getFileById($folder, $id) {
 	 * @return Folder
 	 */
 	private function getFolderForUser($userId) {
-		$path = '/' . $userId . '/files/Notes';
-		if ($this->root->nodeExists($path)) {
-			$folder = $this->root->get($path);
-		} else {
-			$folder = $this->root->newFolder($path);
+		$notesRoot = $this->config->getUserValue($userId, 'notes', 'notesRoot', 'Notes');
+		$userFolder = $this->root->getUserFolder($userId);
+
+		if (!$userFolder->nodeExists($notesRoot)) {
+			return $userFolder->newFolder($notesRoot);
+		}
+
+		$folder = $userFolder->get($notesRoot);
+		$isFolder = $folder->getType() === 'dir';
+		$isShared = $folder->isShared();
+		if (!$isFolder || $isShared) {
```

Comment:
What happens if a user already has some Notes in a folder called Notes, and they (for some reason) have shared the "Notes" folder to someone else?
A new "Notes (2)" folder might be created. And then what happens to the content that was in the "Notes" folder?

Reply:
No. `isShared` is only true if it is an incoming share.

Reply:
Haha, I had the same thought.
But I can confirm isShared() works as Thomas expects, not as we expected.